pysb/core.py

```diff
@@ -1444,7 +1460,7 @@ def enable_synth_deg(self):
         source_cp = as_complex_pattern(self.monomers['__source']())
         if self.compartments:
             for c in self.compartments:
-                source_cp = source_cp ** c
+                source_cp.compartment = c
```

Comment:
OK. One issue I notice here is that each time through the loop, the same CP object is being modified in place then used as the IC pattern. The code path for adding an IC doesn't copy the pattern, so I would think the ICs will actually all be added using the same pattern. This should trigger an exception in `initial` so I'm not confident I'm reading this right. When I get back to my desk I'll investigate further.